classes/announcement/Repository.php

```diff
@@ -161,4 +196,132 @@ public function deleteMany(Collector $collector)
             $this->delete($announcement);
         }
     }
+
+    /**
+     * The subdirectory where announcement images are stored
+     */
+    public function getImageSubdirectory(): string
+    {
+        return 'announcements';
+    }
+
+    /**
+     * Get the base URL for announcement file uploads
+     */
+    public function getFileUploadBaseUrl(?Context $context = null): string
+    {
+        return join('/', [
+            Application::get()->getRequest()->getPublicFilesUrl($context),
+            $this->getImageSubdirectory(),
+        ]);
+    }
+
+    /**
+     * Handle image uploads
+     *
+     * @throws StoreTemporaryFileException Unable to store temporary file upload
+     */
+    protected function handleImageUpload(Announcement $announcement): void
```

Comment:
I think we're missing server-side protection against uploads that aren't images. We're not super tidy about this, but see e.g. `PKPUploadPublicFileHandler` for a series of checks.

Reply:
Can you say a little bit more about what you're expecting here? All of the forms for context settings use the temporary files API endpoint, which is not restricted to images. For example, journal managers can upload a custom CSS file.  Only journal managers or site admins can write to the announcements API endpoint, so they would be trusted users. 

Did you want  the Announcements API handler to return an error if the temporary file is not an image? That would only cover announcement use cases (I'd need to implement the same thing for the highlights/features PRs). Would you rather a helper function in `APIHandler` or the `TemporaryFileManager`? It feels like a more general problem but I'm hesitant to let the scope creep too much here.

Reply:
I agree that this needs general review. I *think* we have a front-end mechanism that checks the upload types in the upload control, but we also need to enforce it on the back end for folks that circumvent that. I would be happy with a quick check at the time of promotion from temporary file that throws an exception.

Reply:
Ok, I've added some checks here: https://github.com/pkp/pkp-lib/pull/9255/commits/a61ed40f8afc7588e4c5179e407daf4ab93dbcab

(Ignore the changes to `PKPPublicFileManager.php`, this was cleanup after a rebase.)